bom/runtime/pom.xml

```diff
@@ -2762,6 +2775,32 @@
                 <artifactId>quarkus-banner</artifactId>
                 <version>${project.version}</version>
             </dependency>
+
+            <!-- reflections -->
+            <dependency>
```

Comment:
I am pretty sure we don't want to add this dependency to the core. We have basically done without it for all this time, so I don't really think it's warranted to add it now.

Reply:
Hi @geoand - if you check the code the reflection framework is used in more than one place. I've just put it on more centralized place. Anyway - what I've tried to achieve here is to register for reflection all classes in given package. Could you please advice is there an other way to achieve it?

Reply:
We use Jandex instead to query the application classes structures, see the examples in other extensions

Reply:
@tpenakov like @gastaldi said, we usually don't use reflection in Quarkus extensions but instead rely on Jandex (which is effectively an index for Java code) to look up various metadata.

One question, do you really need to register all classes of a package for reflection?

Reply:
Hi @geoand ,

The package which I want to register contains the implementation of components for parsing docx format (org.openxmlformats.schemas.wordprocessingml.x2006.main.impl). Also will need to include all classes in packagess for parsing xlsx and pptx components. 
I've tried to google this for Jandex, but without success. If you know how to do it in Jandex (to register all classes for given package) - this will save me a lot of time and I will remove the reflection framework  :)

Reply:
I don't think Jandex does this TBH. If you must register all classes in a package for reflection, then sure go ahead and use the reflections module (if @sberyozkin agrees), but please only use it in your extension and no where else

Reply:
Hi @geoand , @gastaldi  and @sberyozkin,
Reflections maven artifact is now moved under the tika-deployment.